pinot-spi/src/main/java/org/apache/pinot/spi/data/readers/PrimaryKey.java

```diff
@@ -37,7 +41,57 @@ public Object[] getValues() {
   }
 
   public byte[] asBytes() {
-    return SerializationUtils.serialize(_values);
+    int sizeInBytes = 0;
+    byte[][] cache = new byte[_values.length][];
```

Comment:
In majority of the use cases, primary key has only one value, and we can optimize that case by doing one pass